Convert app to use one spinner attached to JFScene + resize spinner

cewert opened the pull request:
## Changes
- Use booleans on main JFScene to control loading spinner
  - `isLoading` and `disableRemote`
- Refactor from 3 to 2 main functions to control spinner
  - `startLoadingSpinner(disableRemote = true as boolean)`
    - Disables the remote as needed with a progressdialog and shows the loading spinner
    - Only display spinner when `isLoading` = false. This prevents double spinners.
  - `stopLoadingSpinner()`
    - Removes progressdialogs as needed and hides the loading spinner
- Modifies every main screne to call `stopLoadingSpinner()` after the data has been loaded and the screne is ready to be shown
- Removed custom Spinner components from all scenes and converted them to use the JFScene spinner
- Resize spinner image from 200px -> 125px and recenter for 1080p resolution. This makes our spinner a similar size and location as the video player spinner
- Rearrange things a bit in `CreateSeasonDetailsGroupByID()` so that the loading spinner goes away when the episodes are loaded instead of waiting on the extras/specials logic to finish

cewert commented:
I pretty much redid this whole PR and updated the OP. Added the main logic to JFScene, using 2 booleans on JFScene instead of the 4 component functions, and I added a `stopLoadingSpinner()` to every main screne when it was done loading data. I also converted all the current custom spinners to use the new JFScene one instead. Our whole app uses one spinner now that can be toggled by using `startLoadingSpinner()` and `stopLoadingSpinner()`.

1hitsong commented:
There's still an old spinner ref on line 553 of `components\ItemGrid\MusicLibraryView.bs`

1hitsong commented:
Inside `Main.bs` there are still 5 references to m.spinner. The ones without isValid checks are throwing errors, then ones with isValid checks simply aren't showing spinners.

cewert commented:
## Changes
- Addressed reviewer feedback
- Fixed all remaining custom spinner references/bugs
- Added the spinner to a few more things triggered from the main loop (selecting a series, season, album etc.)
- Stop showing the spinner a little earlier on audio player view.

1hitsong commented:
I have 1 photo in my photo library. Clicking on it results in an infinite spinner that never goes away.

1hitsong commented:
I pressed Play on the Recordings Library. Nothing loaded and I'm again stuck on an infinite spinner.

sevenrats commented:
this now does fix the missing spinner, but that spinner is behind some darkening left over from the progressDialog for some reason.  This is not the end of the world but it would be better to fix it somehow.
cewert when you have some time, please let me know how you feel about a solution like this one:
https://github.com/cewert/jellyfin-roku/pull/6/files

sevenrats commented:
also just noticed that there is still no spinner between episodes

1hitsong commented:
> I pressed Play on the Recordings Library. Nothing loaded and I'm again stuck on an infinite spinner.

This issue is still occurring. All my other issues are resolved.

cewert commented:
## Changes
- Running `stopLoadingSpinner()` whenever quickplay api quieries are invalid or there are 0 items returned.
- Don't assume item type - Copy recording library logic fix from refactored new function in  #1531

1hitsong commented on source/utils/quickplay.bs:
```diff
@@ -545,7 +571,15 @@ namespace quickplay
             if isValid(tvshowsData) and isValidAndNotEmpty(tvshowsData.items)
-                quickplay.multipleSeries(tvshowsData.items)
+                ' the type of media returned from api may change.
+                if tvshowsData.items[0].Type = "Series"
```
```suggestion
                if LCase(tvshowsData.items[0].Type) = "series"
```
